packages/typespec-ts/src/modular/helpers/operationHelpers.ts

```diff
@@ -498,39 +496,6 @@ function getOptionalWithoutDefault(
   return `"${param.restApiName}": options?.${param.clientName}`;
 }
 
-type OptionalWithDefaultType = (Parameter | Property) & {
-  type: { optional: true; clientDefaultValue: string };
-};
-function isOptionalWithDefault(
-  param: Parameter | Property
-): param is OptionalWithDefaultType {
-  return Boolean(param.clientDefaultValue);
-}
-
-function getOptionalWithDefault(
-  param: OptionalWithDefaultType,
-  importSet: Map<string, Set<string>>
-) {
-  if (param.type.type === "model") {
-    return `"${param.restApiName}": {${getRequestModelMapping(
-      param.type,
-      "options?." + param.clientName,
-      importSet
-    ).join(", ")}} ?? ${getQuotedValue(param)}`;
-  }
-  return `"${param.restApiName}": options.${
-    param.clientName
-  } ?? ${getQuotedValue(param)}`;
-}
-
-function getQuotedValue(param: OptionalWithDefaultType) {
-  if (param.type.type === "string") {
-    return `"${param.clientDefaultValue}"`;
-  } else {
-    return param.clientDefaultValue;
-  }
-}
```

Comment:
Removed the default value logic, because:
1. Mike mentioned we should not set any default value at the client side. They are just for documentaion.
2. The current hlc core client won't set the default value unless it's constant, even if it has logic to handle the default value for required parameters or properties, but we actually don't allow customers to leave a required parameters or properties as empty. 
3. Potential breaking caused by default value changes as suggested by https://github.com/Azure/autorest.csharp/issues/3560

Reply:
Thanks for the clarification! It makes sense to me especially removing the default value could cause unnecessary breaking is risky. 

Maybe not in this pr but I think we could add comment to describe the default value.